### DISCUSSION TITLE
Cloudinary Image Upload // ERROR missing src and POST 500

### DISCUSSION THREAD

**Discussion by Julia-Pickel** [neuefische/questions#784] [Web] [answered]
Hello everyone! 🙌 

We are currently working on the image upload via cloudinary. 
Before we used a simple input field for copy/paste an image-URL (from unplash) in our `NewOutivityForm`-component on our create page. Now we wanna replace this by using the image upload. Therefore we implemented a file-input-field and by submitting the form a new _Outivity_ is created. Unfortunately the image is not displayed then. Neither on the homepage nor on the details page.

![Klein (Bildschirm­foto 2023-11-27 um 11 49 38)](https://github.com/neuefische/questions/assets/145296722/18333c71-b376-4849-8d2b-0c71c3084ebb)

**We're getting an error message in the console after adding a new _Outivity_:**

> <img width="1001" alt="Bildschirm­foto 2023-11-25 um 19 56 21" src="https://github.com/neuefische/questions/assets/145296722/728d328b-8974-4744-aa41-11bc353726f8">

**By creating the new _Outivity_ the new object has an empty image-object.** So I guess it is not correctly passed to our _newOutivity_ as the upload on the cloudinary dashboard works:

> <img width="1007" alt="Bildschirm­foto 2023-11-25 um 19 56 44" src="https://github.com/neuefische/questions/assets/145296722/92624462-54be-43da-955d-b89b00f6aae4">

I have committed and pushed my current status. [It is accessible at this link.](https://github.com/BirkBee/outivities-app/tree/US6_Image-upload) The branch is called _"US6_Image-upload"_.

**What we've already tried:**
We have already searched online for this error and we tried rewriting the `newOutivity`-component. We also double-checked if we defined our user data of cloudinary correctly in the `env.local` and if we set up everything right in the _pages/api/upload/index.js_. However, this error keeps coming up 😢

Hope someone can help – thank you!

**Comment by Roland-Hufnagel** [comment]
Hi Julia.
Please have a look at your async function ``createOutivity(event)``. 
I think there is something missing like:
```js
const image=await response.json();
setSelectedImage(image);
```

**Reply by Julia-Pickel** [reply]
Thank you for your respond and your suggestion Roland!

I've updated the async function (see also the [US6-branch](https://github.com/BirkBee/outivities-app/tree/US6_Image-upload)) but it's still not working. Seems there is an issue with the server-side code handling the upload request.

**By clicking the _save_-button on the _New Outivity_ page I get some errors in the console**: 
(now I even get stuck here and there's no _New Outivity_ created)

![Bildschirm­foto 2023-11-27 um 15 16 42](https://github.com/neuefische/questions/assets/145296722/1aefa728-21ad-43c2-8d75-6a5f3955584e)

![Bildschirm­foto 2023-11-27 um 15 17 12](https://github.com/neuefische/questions/assets/145296722/39d1e2bd-434f-43f7-8814-20f1d79dc8f4)

![Bildschirm­foto 2023-11-27 um 15 17 32](https://github.com/neuefische/questions/assets/145296722/f80b4e87-11ce-4622-9fe2-fb353bc0580f)

![Bildschirm­foto 2023-11-27 um 15 17 45](https://github.com/neuefische/questions/assets/145296722/82824001-5fe3-4c42-849f-de2b3442dbb6)

![Bildschirm­foto 2023-11-27 um 15 20 04](https://github.com/neuefische/questions/assets/145296722/6ba05902-3254-477c-8129-ffeb4abcac5d)

**Comment by Roland-Hufnagel** [comment, marked as answer]
HI Team Outivities,

please try the following:
- in pages/api/upload/index.js Line 24: 
Since the name of the input-field is ``outivityImage`` you must use it here: ``const file = files.outivityImage[0]`` .
- in create.js:
when you have the response you get the image object with ``const image = response.json()`` and then you have to set your useState with ``setSelectedImage(image.secure_url);``. Log out the image variable and you will see that the ``secure_url`` is the link to the image. The link starts with ``res.cloudinary.com/...`` so don't forget to add this URL to your next.config
- in create.js:
When you define the ``newOutivity`` you may want to set ``image: image.secure_url`` because this is the URL to your image
- in NewOutivityForm/index.js:
Since ``selectedImage`` now holds the URL to the image you can use ist here: Instead of ``src={URL.createObjectURL(selectedImage)}`` use ``src={selectedImage}``

**Reply by Julia-Pickel** [reply]
Hi Roland,

many thanks for your detailed answer!
After implementing your suggested changes, everything now works 🎉 

I just made a few more changes in the _NewOutivityForm/index.js_.
As we have a small image preview of the selected image, I had to make some modifications here to display it on the form but also to pass it correctly.

![Bildschirm­foto 2023-11-28 um 10 09 15](https://github.com/neuefische/questions/assets/145296722/936d8d50-f28d-4839-82f6-a598380368fe)
